Makefile

```diff
@@ -39,6 +39,27 @@ $(BUILD_DIR)/util.o: $(LIB_DIR)/util.c $(LIB_DIR)/util.h
 $(BUILD_DIR)/collections.o: $(LIB_DIR)/collections.c $(LIB_DIR)/collections.h
 	$(CC) $(CFLAGS) -c $< -o $@
 
+$(BUILD_DIR)/vec_deque.o: $(LIB_DIR)/vec_deque.c $(LIB_DIR)/vec_deque.h
+	$(CC) $(CFLAGS) -c $< -o $@
+
+# Tests section
```

Comment:
We can also leverage Unity to help us check for memory leaks, I think.  This is lifted from an Exercism `makefile`, but looks like something we can modify to our benefit:

```
...
CFLAGS += -DUNITY_SUPPORT_64 -DUNITY_OUTPUT_COLOR

ASANFLAGS  = -fsanitize=address
ASANFLAGS += -fno-common
ASANFLAGS += -fno-omit-frame-pointer

.PHONY: test
test: tests.out
	@./tests.out

.PHONY: memcheck
memcheck: ./*.c ./*.h
	@echo Compiling $@
	@$(CC) $(ASANFLAGS) $(CFLAGS) test-framework/unity.c ./*.c -o memcheck.out $(LIBS)
	@./memcheck.out
	@echo "Memory check passed"

.PHONY: clean
clean:
	rm -rf *.o *.out *.out.dSYM

tests.out: ./*.c ./*.h
	@echo Compiling $@
	@$(CC) $(CFLAGS) test-framework/unity.c ./*.c -o tests.out $(LIBS)
```

Reply:
ooo that's a good idea!